lib/active_job_status/hooks.rb

```diff
@@ -2,12 +2,54 @@ module ActiveJobStatus
   module Hooks
     def self.included(base)
       base.class_eval do
-        before_enqueue { job_tracker.enqueued }
 
-        before_perform { job_tracker.performing }
+        around_enqueue do |job, block|
+          begin
+            before_enqueue(job)
+            job_tracker.enqueued
+            block.call
+            on_enqueue_success(job)
+          rescue Exception => exception
+            job_tracker.failed
+            on_enqueue_failure(exception, job)
+            raise
```

Comment:
I'm a bit concerned that we might be swallowing useful information here. Perhaps I'm mistaken, but if a user does not supply a custom `on_enqueue_failure` block, all exceptions will be caught, and the actual exception will get swallowed, and line 15 will raise a basic exception instead.